File: gims_mobile_app/lib/models/marine/marine_money_receipt.dart
import 'marine_bill.dart';

class MarineMoneyReceipt {
  int? id;
  String? issuingOffice;
  String? classOfInsurance;
  DateTime? date;
  String? modeOfPayment;
  String? issuedAgainst;
  MarineBill? marinebill;

  MarineMoneyReceipt({
    this.id,
    this.issuingOffice,
    this.classOfInsurance,
    this.date,
    this.modeOfPayment,
    this.issuedAgainst,
    this.marinebill,
  });

  MarineMoneyReceipt.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        issuingOffice = json['issuingOffice'],
        classOfInsurance = json['classOfInsurance'],
        date = json['date'] != null ? DateTime.tryParse(json['date'].toString()) : null,
        modeOfPayment = json['modeOfPayment'],
        issuedAgainst = json['issuedAgainst'],
        marinebill = json['marinebill'] != null
            ? MarineBill.fromJson(json['marinebill'])
            : null;

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    if (id != null) data['id'] = id;
    data['issuingOffice'] = issuingOffice;
    data['classOfInsurance'] = classOfInsurance;
    if (date != null) data['date'] = date?.toIso8601String();
    data['modeOfPayment'] = modeOfPayment;
    data['issuedAgainst'] = issuedAgainst;
    if (marinebill != null) {
      data['marinebill'] = {'id': marinebill!.id};
    }
    return data;
  }
}

File: gims_backend_api/src/main/java/com/kutub/InsuranceManagement/entity/marine/MarineMoneyReceipt.java
package com.kutub.InsuranceManagement.entity.marine;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "marine_money_receipt")
public class MarineMoneyReceipt {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "issueOfficeId")
    private com.kutub.InsuranceManagement.entity.utility.IssueOffice issuingOffice;

    private String classOfInsurance;

    @Temporal(TemporalType.DATE)
    private Date date = new Date();

    private String modeOfPayment;

    private String issuedAgainst;

    @ManyToOne(fetch = FetchType.EAGER, cascade = { CascadeType.PERSIST, CascadeType.MERGE })
    @JoinColumn(name = "marinebillId")
    private MarineBill marinebill;

}
